src/shogun/classifier/svm/WDSVMOcas.cpp

```diff
@@ -156,21 +156,15 @@ int32_t CWDSVMOcas::set_wd_weights()
 	return w_dim_single_c;
 }
 
-bool CWDSVMOcas::train_machine(CFeatures* data)
+template <typename ST>
+bool CWDSVMOcas::train_machine_templated(CStringFeatures<ST>* data)
 {
 	SG_INFO("C=%f, epsilon=%f, bufsize=%d\n", get_C1(), get_epsilon(), bufsize)
 
 	ASSERT(m_labels)
 	ASSERT(m_labels->get_label_type() == LT_BINARY)
-	if (data)
-	{
-		if (data->get_feature_class() != C_STRING ||
-				data->get_feature_type() != F_BYTE)
-		{
-			SG_ERROR("Features not of class string type byte\n")
-		}
-		set_features((CStringFeatures<uint8_t>*) data);
-	}
+
+	set_features((CStringFeatures<uint8_t>*)data);
```

Comment:
this will fail if you pass anythign else then uin8_t string features. You will need to assert the above if you want to rely on this: https://github.com/shogun-toolbox/shogun/pull/4373/files#diff-2a3c04af057ddaaf9cfa245feef22d22L167